core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -430,58 +432,87 @@ type MaybeMessage<Block> = Option<(Vec<PeerId>, NeighborPacket<NumberFor<Block>>
 impl<Block: BlockT> Inner<Block> {
 	fn new(config: crate::Config) -> Self {
 		Inner {
-			local_view: View::default(),
+			local_view: None,
 			peers: Peers::default(),
 			live_topics: KeepTopics::new(),
 			next_rebroadcast: Instant::now() + REBROADCAST_AFTER,
+			authorities: Vec::new(),
 			config,
 		}
 	}
 
-	/// Note a round in a set has started.
-	fn note_round(&mut self, round: Round, set_id: SetId) -> MaybeMessage<Block> {
-		if self.local_view.round == round && self.local_view.set_id == set_id {
-			return None;
-		}
+	/// Note a round in the current set has started.
+	fn note_round(&mut self, round: Round) -> MaybeMessage<Block> {
+		{
```

Comment:
Is the new scope necessary?

Reply:
yes, because `self.local_view` is borrowed mutably within it. So we wouldn't be able to call another function that borrows `self`.